### PR TITLE
Updated and renamed spark-client.md to spark-metadata-client.md

### PR DESCRIPTION
**Pull request by jimmymcnutty**
Closes #(Insert issue number closed by this PR)

### Background

Share context and relevant information for the PR: offline discussions, considerations, design decisions etc.

### Bug Fix

If this PR is a bug fix, please let us know about:

1. Problem - The reason for opening the bug
2. Root cause - Discovered root cause after investigation
3. Solution - How the bug was fixed
      
### New Feature

If this PR introduces a new feature, describe it here.

### Testing Details

How were the changes tested?

### Breaking Change?

Does this change break any existing functionality? (API, CLI, Clients)

## Additional info

Logs, outputs, screenshots of changes if applicable (CLI / GUI changes)

### Contact Details

How can we get in touch with you if we need more info? (ex. email@example.com)

### CODE REVIEW

File: docs/reference/spark-metadata-client.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ---
-title: Spark Client
-description: The lakeFS Spark client performs operations on lakeFS committed metadata stored in the object store. 
+title: Spark Metadata Client
+description: The lakeFS Spark client performs operations on lakeFS committed Metadata stored in the object store.
```
Comment (ozkatz) [inline docs/reference/spark-metadata-client.md:3]: Why captitalize it everywhere?

**Review by ozkatz — CHANGES_REQUESTED** [review]
Thanks!

**Comment by N-o-Z** [comment]
@jimmymcnutty  hey - we should either fix the comments and merge or close this PR

**Comment by N-o-Z** [comment]
Closing due to inactivity

**Pull request closed without merging**